app/components/file/report-drawer/va-reports/index.ts

```diff
@@ -0,0 +1,273 @@
+    let btnLabel = this.intl.t('generateReport');
+
+    if (this.latestReportIsGenerating || this.isReportGenerating) {
+      btnLabel = this.intl.t('generatingReport');
+    } else if (this.canGenerateReport) {
+      btnLabel = this.intl.t('generateReport');
+    }
+
+    return btnLabel;
+  }
+
+  get prevReports() {
+    const noOfReports = Number(this.reports?.length);
+
+    if (
+      this.canGenerateReport &&
+      !this.latestReportIsGenerating &&
+      noOfReports === 1
+    ) {
+      return this.reports;
+    } else if (
+      this.canGenerateReport &&
+      !this.latestReportIsGenerating &&
+      noOfReports > 1
+    ) {
+      return this.reports?.slice(0, REPORT.MAX_LIMIT - 1);
+    } else if (noOfReports > 1) {
+      return this.reports?.slice(1, REPORT.MAX_LIMIT);
+    }
+
+    return [];
+  }
+
+  get previousReport() {
+    return this.prevReports?.firstObject;
+  }
+
+  get reportPassword() {
+    return this.file.project.get('pdfPassword');
+  }
+
+  get reportDetails(): FileReportDetails[] {
+    return [
+      {
+        fileReport: this.latestReport,
+        type: 'pdf' as const,
+        primaryText: `${this.intl.t('fileReport.detailedReport', {
+          reportType: 'pdf',
+        })}`,
+        secondaryText: `
+        ${this.intl.t('fileReport.reportPassword', {
+          password: this.reportPassword,
+        })}`,
+        hideReport: this.showGenerateReportBtn,
+        copyText: this.reportPassword,
+        iconComponent: 'ak-svg/pdf-report' as const,
+      },
+      {
+        fileReport: this.latestReport,
+        type: 'xlsx' as const,
+        primaryText: `${this.intl.t('fileReport.summaryReport', {
+          reportType: 'xlsx',
+        })}`,
+        secondaryText: `${this.intl.t('fileReport.noPasswordRequired')}`,
+        hideReport: this.showGenerateReportBtn,
+        iconComponent: 'ak-svg/xlsx-icon' as const,
+      },
+      {
+        fileReport: this.latestReport,
+        type: 'csv' as const,
+        primaryText: `${this.intl.t('fileReport.summaryReport', {
+          reportType: 'csv',
+        })}`,
+        secondaryText: `${this.intl.t('fileReport.noPasswordRequired')}`,
+        hideReport: this.showGenerateReportBtn,
+        iconComponent: 'ak-svg/csv-icon' as const,
+      },
+      {
+        fileReport: this.previousReport,
+        type: 'pdf' as const,
+        primaryText: `${this.intl.t('fileReport.previousScan', {
+          reportType: 'pdf',
+        })}`,
+        secondaryText: `
+        ${this.intl.t('fileReport.reportPassword', {
+          password: this.reportPassword,
+        })}`,
+        hideReport: !this.previousReport,
+        copyText: this.reportPassword,
+        iconComponent: 'ak-svg/pdf-report' as const,
+        generatedOnDateTime: this.previousReport?.generatedOnDateTime,
+      },
+    ];
+  }
+
+  // Actions
+  @action
+  observeReportCounter() {
+    this.getReports.perform();
+  }
+
+  @action initializeDrawer() {
+    addObserver(
```

Comment:
do we require this observer ? `getReports` will anyways be called when drawer opens

Reply:
Yeah. This is required for the Websocket connection that triggers an api call to get the latest report status.